config/user-compiler-options.m4

```diff
@@ -0,0 +1,68 @@
+dnl #
+dnl # Check if gcc supports -fsanitize=address. Applies to all libraries,
+dnl # ztest and zdb
+dnl #
+dnl # LDFLAGS needs -fsanitize=address at all times so libraries compiled with
+dnl # it will be linked successfully. CFLAGS will vary by binary being built.
+dnl #
+AC_DEFUN([ZFS_AC_CONFIG_USER_GCC_FSANITIZE], [
+	if test "$enable_debug" = "yes"
+	then
+		AC_MSG_CHECKING([for -fsanitize=address support])
+
+		saved_cflags="$CFLAGS"
+		saved_ldflags="$LDFLAGS"
+		CFLAGS="$CFLAGS -fsanitize=address"
+		LDFLAGS="$LDFLAGS -fsanitize=address"
+
+		AC_TRY_LINK(
+		[
+			#include <time.h>
+		],[
+			time_t now;
+			now = time(&now);
+		],[
+			COMPILER_FSANITIZE_ADDRESS=-fsanitize=address
+			saved_ldflags="$saved_ldflags -fsanitize=address"
+			AC_MSG_RESULT(yes)
+		],[
+			COMPILER_FSANITIZE_ADDRESS=""
+ 			AC_MSG_RESULT(no)
+		])
+
+		CFLAGS="$saved_flags"
+		LDFLAGS="$saved_ldflags"
+	fi
+	AC_SUBST([COMPILER_FSANITIZE_ADDRESS])
```

Comment:
I'd suggest either dropping the `COMPILER_` prefix, which isn't super useful and would be my preference.  Or alternately adding it to NO_FORMAT_TRUNCATION and FRAME_LARGER_THAN for consistency.